yt_dlp/extractor/medialaan.py

```diff
@@ -75,51 +194,32 @@ class MedialaanIE(InfoExtractor):
             'id': '1576607',
             'ext': 'mp4',
             'title': 'Tom Waes blaastest',
+            'channel': 'De Morgen',
+            'channel_id': '352',
+            'description': 'Tom Waes werkt mee aan een alcoholcampagne op Werchter',
             'duration': 62,
+            'genres': ['News'],
+            'release_date': '20250705',
+            'release_timestamp': 1751730795,
+            'series': 'Nieuwsvideo\'s',
+            'series_id': '1683',
+            'tags': 'count:1',
             'thumbnail': r're:https?://video-images\.persgroep\.be/aws_generated.+\.jpg',
-            'timestamp': 1751730795,
-            'upload_date': '20250705',
+            'uploader': 'De Morgen',
+            'uploader_id': '17',
         },
         'params': {'extractor_args': {'generic': {'impersonate': ['chrome']}}},
     }]
 
     @classmethod
     def _extract_embed_urls(cls, url, webpage):
-        entries = []
-        for element in re.findall(r'(<div[^>]+data-mychannels-type="video"[^>]*>)', webpage):
-            mychannels_id = extract_attributes(element).get('data-mychannels-id')
-            if mychannels_id:
-                entries.append('https://mychannels.video/embed/' + mychannels_id)
-        return entries
+        return traverse_obj(webpage, (
+            {find_elements(tag='div', attr='data-mychannels-type', value='video', html=True)},
+            ..., {extract_attributes}, 'data-mychannels-id', {str},
+            {lambda x: f'https://mychannels.video/embed/{x}'}, {url_or_none}, filter, all, filter,
+        ))
```

Comment:
we can `yield from` here

and more importantly, we need to `filter` before the f-string transformation or else we could end up with `/embed/None` URLs

and since we're filtering beforehand (and not using `all`) that means everything after the transformation is unnecessary
```suggestion
        yield from traverse_obj(webpage, (
            {find_elements(tag='div', attr='data-mychannels-type', value='video', html=True)},
            ..., {extract_attributes}, 'data-mychannels-id', {str}, filter,
            {lambda x: f'https://mychannels.video/embed/{x}'}))
```